apps/emqx_ft/test/emqx_ft_conf_SUITE.erl

```diff
@@ -85,3 +86,86 @@ t_update_config(_Config) ->
         5 * 60 * 1000,
         emqx_ft_conf:gc_interval(emqx_ft_conf:storage())
     ).
+
+t_remove_restore_config(Config) ->
+    ?assertMatch(
+        {ok, _},
+        emqx_conf:update([file_transfer, storage], #{<<"type">> => <<"local">>}, #{})
+    ),
+    ?assertEqual(
+        60 * 60 * 1000,
+        emqx_ft_conf:gc_interval(emqx_ft_conf:storage())
+    ),
+    % Verify that transfers work
+    ok = emqx_ft_test_helpers:upload_file(gen_clientid(), <<"f1">>, "f1", <<?MODULE_STRING>>),
+    ?assertMatch(
+        {ok, _},
+        emqx_conf:remove([file_transfer, storage], #{})
+    ),
+    ?assertEqual(
+        undefined,
+        emqx_ft_conf:storage()
+    ),
+    ?assertEqual(
+        undefined,
+        emqx_ft_conf:gc_interval(emqx_ft_conf:storage())
+    ),
+    ClientId = gen_clientid(),
+    Client = emqx_ft_test_helpers:start_client(ClientId),
+    % Verify that transfers fail cleanly when storage is disabled
+    ?check_trace(
+        ?assertMatch(
+            {ok, #{reason_code_name := unspecified_error}},
+            emqtt:publish(
+                Client,
+                <<"$file/f2/init">>,
+                emqx_json:encode(emqx_ft:encode_filemeta(#{name => "f2", size => 42})),
+                1
+            )
+        ),
+        fun(Trace) ->
+            ?assertMatch(
+                [#{transfer := {ClientId, <<"f2">>}, reason := {error, disabled}}],
+                ?of_kind("store_filemeta_failed", Trace)
+            )
```

Comment:
This is also imperfect IMO. I'm currently thinking that ideally we'd want to give users an ability to "turn off" file transfer feature, which in turn would just remove relevant `emqx_ft` hooks.
Q: Do you think the same? How do we do that "turning off" given that `emqx_conf:remove([file_transfer])` is prohibited? Special `enabled` flag in the config schema?

Reply:
Made a global killswitch, see https://github.com/emqx/emqx/pull/10501/commits/d190398b51861dc6d2a930cdb4f61b67b66ebcfd.